programs/fileio.c

```diff
@@ -1782,6 +1782,54 @@ FIO_compressFilename_srcFile(FIO_ctx_t* const fCtx,
     return result;
 }
 
+#ifdef __GNUC__
+static const char* checked_index(const char* options[], size_t length __attribute__((unused)), size_t index)
```

Comment:
minor: for such a warning, prefer adding `(void)length;` somewhere in the code below.
It's a portable way to express that a variable may not be used and it's intentional.
It avoids usage of compiler-specific extensions.